server/src/main/java/io/druid/server/coordination/ZkCoordinator.java

```diff
@@ -349,13 +349,11 @@ each time when addSegment() is called, it has to wait for the lock in order to m
         }
       }
       loadSegment(segment, callback);
-      if (!announcer.isAnnounced(segment)) {
-        try {
-          announcer.announceSegment(segment);
-        }
-        catch (IOException e) {
-          throw new SegmentLoadingException(e, "Failed to announce segment[%s]", segment.getIdentifier());
-        }
+      try {
+        announcer.announceSegment(segment);
```

Comment:
updated BatchDataSegmentAnnouncer.announce(..) to always do the check so that it gets done in other places as well.